src/runtime/virtcontainers/clh.go

```diff
@@ -820,14 +888,29 @@ func (clh *cloudHypervisor) ResizeMemory(ctx context.Context, reqMemMB uint32, m
 	ctx, cancelResize := context.WithTimeout(ctx, clhAPITimeout*time.Second)
 	defer cancelResize()
 
-	resize := *chclient.NewVmResize()
-	// OpenApi does not support uint64, convert to int64
-	resize.DesiredRam = func(i int64) *int64 { return &i }(int64(newMem.ToBytes()))
-	clh.Logger().WithFields(log.Fields{"current-memory": currentMem, "new-memory": newMem}).Debug("updating VM memory")
-	if _, err = cl.VmResizePut(ctx, resize); err != nil {
-		clh.Logger().WithError(err).WithFields(log.Fields{"current-memory": currentMem, "new-memory": newMem}).Warnf("failed to update memory %s", openAPIClientError(err))
-		err = fmt.Errorf("Failed to resize memory from %d to %d: %s", currentMem, newMem, openAPIClientError(err))
-		return uint32(currentMem.ToMiB()), MemoryDevice{}, openAPIClientError(err)
+	if !info.Config.Memory.HasZones() {
+		// Resize memory when no using memory zone, hotplug method can be acpi or virtio-mem
+		resize := *chclient.NewVmResize()
+		// OpenApi does not support uint64, convert to int64
```

Comment:
@fidencio noticed this recently too. Do we know why they still don't support this common data type?

Reply:
I assume it's because JS doesn't have a concept of unsigned integers.

Reply:
Ugh - that sounds like a good source of bugs and confusion to me! `openapi-rs` anyone? :smile: